zorg/buildbot/builders/annotated/rise-riscv-gauntlet-build.sh

```diff
@@ -0,0 +1,162 @@
+  -DLLVM_LIT_ARGS="-v" \
+  -DCMAKE_C_COMPILER=clang \
+  -DCMAKE_CXX_COMPILER=clang++ \
+  -DLLVM_ENABLE_LLD=True \
+  -DLLVM_TARGETS_TO_BUILD="RISCV;X86" \
+  -DCMAKE_C_COMPILER_LAUNCHER=ccache \
+  -DCMAKE_CXX_COMPILER_LAUNCHER=ccache \
+  -DLLVM_ENABLE_PROJECTS="lld;clang;llvm" \
+  -B llvm-project/build/stage1 \
+  -S llvm-project/llvm
+
+build_step "llvm-project build stage 1"
+cmake --build llvm-project/build/stage1
+
+STAGE1_BINDIR=$(pwd)/llvm-project/build/stage1/bin
+
+# Don't exit immediately upon failure from here on.
+set +e
+
+# Skip a few tests that have excessive runtimes relative to the others.
+export LIT_FILTER_OUT='(SingleSource/Benchmarks/Polybench/linear-algebra/solvers/(ludcmp|lu)|MicroBenchmarks/LoopVectorization/LoopInterleavingBenchmarks)'
+for CONF in rva20 rva22 rva23 rva23-evl rva23-mrvv-vec-bits; do
+  RVA23_QEMU_CPU="rv64,zba=true,zbb=true,zbc=false,zbs=true,zfhmin=true,v=true,vext_spec=v1.0,zkt=true,zvfhmin=true,zvbb=true,zvkt=true,zihintntl=true,zicond=true,zimop=true,zcmop=true,zcb=true,zfa=true,zawrs=true,rvv_ta_all_1s=true,rvv_ma_all_1s=true,rvv_vl_half_avl=true"
+  case "$CONF" in
+    rva20)
+      CFLAGS="-march=rva20u64"
+      QEMU_CPU="rv64,zfa=false,zba=false,zbb=false,zbc=false,zbs=false"
+      ;;
+    rva22)
+      CFLAGS="-march=rva22u64"
+      QEMU_CPU="rv64,zba=true,zbb=true,zbc=false,zbs=true,zfhmin=true,v=false,zkt=true,zihintntl=true"
+      ;;
+    rva23)
+      CFLAGS="-march=rva23u64"
+      QEMU_CPU=$RVA23_QEMU_CPU
+      ;;
+    rva23-evl)
+      CFLAGS="-march=rva23u64 -mllvm -force-tail-folding-style=data-with-evl -mllvm -prefer-predicate-over-epilogue=predicate-else-scalar-epilogue"
+      QEMU_CPU=$RVA23_QEMU_CPU
+      ;;
+    rva23-mrvv-vec-bits)
+      CFLAGS="-march=rva23u64 -mrvv-vector-bits=zvl"
+      QEMU_CPU=$RVA23_QEMU_CPU
+      ;;
+    *)
+      echo "Unrecognised config name"
+      exit 1
+  esac
+  export QEMU_LD_PREFIX="$(pwd)/../rvsysroot"
+  export QEMU_CPU=$QEMU_CPU
+  cat - <<EOF > $CONF-toolchain.cmake
+set(CMAKE_SYSTEM_NAME Linux)
+set(CMAKE_SYSROOT $(pwd)/../rvsysroot)
+set(CMAKE_C_COMPILER_TARGET riscv64-linux-gnu)
+set(CMAKE_CXX_COMPILER_TARGET riscv64-linux-gnu)
+set(CMAKE_C_FLAGS_INIT "$CFLAGS -DSMALL_PROBLEM_SIZE")
+set(CMAKE_CXX_FLAGS_INIT "$CFLAGS -DSMALL_PROBLEM_SIZE")
+set(CMAKE_LINKER_TYPE LLD)
+set(CMAKE_C_COMPILER $STAGE1_BINDIR/clang)
+set(CMAKE_CXX_COMPILER $STAGE1_BINDIR/clang++)
+set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
+set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
+set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
+set(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE ONLY)
+EOF
+  build_step "$CONF: llvm-test-suite configure"
+  cmake -G Ninja \
+    --toolchain=$(pwd)/$CONF-toolchain.cmake \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DTEST_SUITE_LIT=$STAGE1_BINDIR/llvm-lit \
+    -DTEST_SUITE_LIT_FLAGS=-v \
+    -DTEST_SUITE_COLLECT_CODE_SIZE=OFF \
+    -DTEST_SUITE_COLLECT_COMPILE_TIME=OFF \
+    -DTEST_SUITE_USER_MODE_EMULATION=ON \
```

Comment:
I presume `binfmt_misc` is setup on the worker somehow?

Reply:
Yes this just relies on binfmt_misc (with the exported QEMU_* variables controlling needed qemu options).